terminus-ui/tabs/src/body/tab-body.component.ts

```diff
@@ -0,0 +1,237 @@
+  host: {
+    class: 'ts-tab-body',
+  },
+  animations: [tsTabsAnimations.translateTab],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsTabBody',
+})
+export class TsTabBodyComponent implements OnInit, OnDestroy {
+  /**
+   * A unique ID per instance
+   */
+  protected id: number = nextUniqueId++;
+
+  /**
+   * Current position of the tab-body in the tab-group. Zero means that the tab is visible.
+   */
+  private positionIndex: number;
+
+  /**
+   * Emits when an animation on the tab is complete
+   */
+  public translateTabComplete = new Subject<AnimationEvent>();
+
+  /**
+   * Computes the position state based on the specified origin position.
+   * This is used if the tab is becoming visible immediately after creation.
+   */
+  private get computedPositionFromOrigin(): TsTabBodyPositionState {
+    return (isNumber(this.origin) && (this.origin < 1)) ? 'left-origin-center' : 'right-origin-center';
+  }
+
+  /**
+   * Determine the computed position state that will be used for the tab-body animation trigger
+   */
+  private get computedPositionAnimationState(): 'left' | 'right' | 'center' {
+    const position = (this.positionIndex < 0)
+      ? 'left'
+      : (this.positionIndex > 0)
+      ? 'right'
+      : 'center'
+    ;
+    return position;
+  }
+
+  /**
+   * The tab body content to display
+   */
+  @Input()
+  public content: TemplatePortal;
+
+  /**
+   * Position that will be used when the tab is immediately becoming visible after creation
+   */
+  @Input()
+  public origin: number;
+
+  /**
+   * The shifted index position of the tab body, where zero represents the active center tab.
+   */
+  @Input()
+  public set position(position: number) {
+    this.positionIndex = position;
+    this.positionState = this.computedPositionAnimationState;
+  }
+
+  /**
+   * Tab body position state.
+   * Used by the animation trigger for the current state.
+   *
+   * NOTE: We cannot use a single getter/setter for position because the types are different
+   */
+  public positionState!: TsTabBodyPositionState;
+
+  /**
+   * Event emitted when the tab begins to animate towards the center as the active tab
+   */
+  @Output()
+  readonly onCentering: EventEmitter<number> = new EventEmitter<number>();
+
+  /**
+   * Event emitted before the centering of the tab begins
+   */
+  @Output()
+  readonly beforeCentering: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  /**
+   * Event emitted after the tab has left the center position
+   */
+  @Output()
+  readonly afterLeavingCenter: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  /**
+   * Event emitted when the tab completes its animation towards the center
+   */
+  @Output()
+  readonly onCentered: EventEmitter<void> = new EventEmitter<void>(true);
+
+
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+    private changeDetectorRef: ChangeDetectorRef,
```

Comment:
Is `changeDetectorRef` required even if you don't call it? Or `.markForCheck` should be called?

Reply:
Ahh good catch.. that was left over from Material where they were calling the cdr when the direction changed